README.md

```diff
@@ -1,15 +1,111 @@
+
+## Usage
+
+### Server CLI
+
+After installing the server and the client apps, start the server client with your preffered options:
+
+```console
+LocalNetAppChat.Server [options]
+
+  Options:
+
+    --listenOn           The IP Address the server should start litening on (e.g localhost)
+    --port               The port the server should connect to (default: 5000)
+    --https              Whether to start the server as HTTPS or HTTP server
+    --key                An Authentication password that the client should send along the requests to be able to perform tasks. (default: 1234)
+
+  Examples:
+
+  – Start the server with the default settings
+    $ LocalNetAppChat.Server
+  - Start the server in HTTPS mode
+    $ LocalNetAppChat.Server --https
+  - Start the server with different ip and port and custom key
+    $ LocalNetAppChat.Server --listenOn "54.15.12.1" --port "54822" --https --key "HeythereGithubExample"
+
+```
+
+### Client CLI
+
+The client can be run in two modes, the `message` and `listener` mode. The `message` mode allows the client to send messages to the server that then will be displayed either to all the clients or only to one client. When running the client in `listener` mode, it will only listen to incoming messages done by other clients from the server.
+
+```console
+LocalNetAppChat.ConsoleClient [options]
+
+  Options:
+
+    message              Run the client in message mode
+    listener             Run the client in listener mode
+    --text               The text message to send to the server. (only when in message mode!)
+    --clientName         The name of your client. If not specified, your machine name will be sent as clientName to the server
+    --ignoresslerrors    Ignore SSL Erros and do not print them in the console
+    --server             The IP Address the server should start litening on (e.g localhost)
+    --port               The port the server should connect to (default: 5000)
+    --https              Whether to start the server as HTTPS or HTTP server
+    --key                An Authentication password that the server requires to allow incoming requests from the client!
+
+  Examples:
+
+  – Start the client in listening mode
+    $ LocalNetAppChat.ConsoleClient listener --server "localhost" --port 54214 --key 1234 --clientName "GithubReadMe"
+  - Start the client in message mode
+    $ LocalNetAppChat.ConsoleClient message --server "localhost" --port 51234 --key 1234 --text "Hey there, I am client GithubReadMe"
+
+```
+
+## ToDo
+
+- [x] add documentation
 - [ ] add task management
   - [ ] is task management another bot?
 - [ ] powershell clients
 - [ ] do message management more correctly (forget them after 1 hour)
-
-## maybe later
-
 - [ ] microsoft teams relay?
 - [ ] discord relay?
 - [ ] performance testing with naseifs typescript implementation
+
+## Contributions
+
+Software contributions are welcome. If you are not a dev, testing and reproting bugs can also be very helpful!
```

Comment:
Please run a spell checker. I think there are several smaller typos that we will catch this way. Although I only saw "reproting" instead of "reporting". Thank you very much!